neo/core/spiketrain.py

```diff
@@ -975,7 +975,7 @@ def _merge_array_annotations(self, others, sorting=None):
 
         omitted_keys_other = [
             key
-            for key in np.unique([key for other in others for key in other.array_annotations])
+            for key in set([key for other in others for key in other.array_annotations])
```

Comment:
only actual change to code base. numpy 2.0 returns this as a `np.str_` thing rather than as the actual string. Using `set` instead of `np.unique` returns the string both >2.0 and < 2.0 for NumPy. Maybe a slight performance hit, but probably not too bad since the annotations should be small either way and this is in a list and not array.